datafusion/optimizer/src/decorrelate_dependent_join.rs

```diff
@@ -484,55 +443,25 @@ impl DependentJoinDecorrelator {
         correlated_map: &IndexMap<Column, Column>,
     ) -> Result<LogicalPlan> {
         // replace correlated column in dependent with delimget's column
-        let new_plan = if let LogicalPlan::DependentJoin(DependentJoin {
-            schema,
-            correlated_columns,
-            subquery_expr,
-            subquery_depth,
-            left,
-            right,
-            subquery_name,
-            lateral_join_condition,
-        }) = plan
-        {
-            let mut new_correlated_columns = vec![];
-
-            for corr in correlated_columns.iter() {
-                let mut col = corr.col.clone();
-                if let Some(delim_col) = correlated_map.get(&corr.col) {
-                    col = delim_col.clone();
-                }
-
-                new_correlated_columns.push(CorrelatedColumnInfo {
-                    col,
-                    data_type: corr.data_type.clone(),
-                    depth: corr.depth,
-                });
-            }
-
-            LogicalPlan::DependentJoin(DependentJoin {
-                schema: schema.clone(),
-                correlated_columns: new_correlated_columns,
-                subquery_expr: subquery_expr.clone(),
-                subquery_depth: subquery_depth.clone(),
-                left: left.clone(),
-                right: right.clone(),
-                subquery_name: subquery_name.clone(),
-                lateral_join_condition: lateral_join_condition.clone(),
-            })
+        let new_plan = if let LogicalPlan::DependentJoin(DependentJoin { .. }) = plan {
+            return internal_err!(
```

Comment:
duckdb allow rewriting for dependent join some how, i need to check if this wil ever happen